### ISSUE TITLE
[BUG] order_flow_imbalance: stop_loss becomes 0.0 with zero/empty price data

### ISSUE THREAD

**Issue by yurisa2** [PetroSa2/petrosa-bot-ta-analysis#296]
## Problem Statement
The `order_flow_imbalance` strategy generates invalid stop-loss values of 0.0 when support/resistance level calculations encounter edge cases with zero or empty price data.

### Error Logs
```
2026-09-18 03:31:58,948 ERROR - order_flow_imbalance: REJECTED - Missing or invalid risk parameters for buy
2026-09-18 03:31:58,947 ERROR - Validation failed for order_flow_imbalance: risk parameters must be positive. SL: 0.0, TP: 40.3515
```

### Root Cause Analysis
In `ta_bot/strategies/order_flow_imbalance.py` lines 88-91 and 96-99:

```python
# For accumulation - SL below recent support
support_level = df["low"].iloc[-20:-10].min()
stop_loss = support_level * 0.99  # 1% below support
```

Issues:
1. If `df["low"]` contains zeros in the slice `[-20:-10]`, `min()` returns 0
2. If the dataframe has fewer than 20 rows, `iloc[-20:-10]` returns an empty Series, and `min()` on empty Series may return NaN or unexpected values
3. The `support_level * 0.99` calculation with 0 yields `stop_loss = 0.0`

### Reproduction Test
See `petrosa-bot-ta-analysis/_bmad-output/incidents/2026-09-18/reproduction_test.py::TestOrderFlowImbalanceZeroSL`

## Acceptance Criteria
- [ ] Add guard checks before using `support_level`/`resistance_level` values
- [ ] Return `None` early if price data slice is empty or contains invalid values (<= 0)
- [ ] Add defensive checks to ensure `stop_loss` is never 0 or negative
- [ ] Add unit tests covering edge cases (zero prices, empty slices, NaN values)
- [ ] All existing tests continue to pass

## Technical Details
**File**: `ta_bot/strategies/order_flow_imbalance.py`
**Lines**: 88-101 (buy case), 96-101 (sell case)
**Error Type**: Runtime - Invalid risk parameter generation

## Related
- Incident: INC-2026-09-18-001
- Evidence: `petrosa-bot-ta-analysis/_bmad-output/incidents/2026-09-18/`

**Comment by yurisa2** [comment]
### MemPalace pre-flight report

**Dispatcher:** ticket-orchestrator
**Ticket:** PetroSa2/petrosa-bot-ta-analysis#296 — "[BUG] order_flow_imbalance: stop_loss becomes 0.0 with zero/empty price data"
**Affected service phrase:** "bot-ta-analysis order_flow_imbalance stop_loss"

**Searches run**
- semantic search on title — 5 hits, top: drawer_petrosa-realtime-strategies_bugs_b909a3f7a260378344bd9d16 (unrelated depth-key bug, similarity 0.47)
- kg query on `PetroSa2/petrosa-bot-ta-analysis#296` — 0 triples
- tunnel discovery on "bot-ta-analysis order_flow_imbalance stop_loss" — 0 tunnels

**Key findings**
- No prior ticket/PR/drawer directly addresses this order_flow_imbalance stop_loss=0.0 defect.
- Related but distinct memory: petrosa_k8s#1057 investigation (2026-09-15) noted order_flow_imbalance (among other strategies) emitting zero live signals — different symptom (silence vs invalid SL), same strategy file family. Worth a mention in PR notes but not a blocker.
- No architectural fact contradicts the issue's root-cause analysis (support/resistance min() on zero/empty slice).

**Decision impact**
- No impact — proceeding as planned with the fix described in the issue body (guard non-positive/empty price slices before min()/max(), skip signal if no valid support/resistance).

**Repo-drift flag:** no (issue_repo == petrosa-bot-ta-analysis, ticket implements code in the same repo)

**MemPalace availability:** OK

**Comment by yurisa2** [comment]
### BMAD lifecycle: work started

Work on this ticket has started. Project board status: **In progress**.

**Comment by yurisa2** [comment]
### BMAD lifecycle: ready for review

Pull request: https://github.com/PetroSa2/petrosa-bot-ta-analysis/pull/300

Project board status: **In review**.

**Comment by yurisa2** [comment]
### BMAD lifecycle: review verdict — **Accept**

PR #300 fix reviewed: adds OrderFlowImbalanceStrategy._safe_level() guard filtering non-positive/NaN prices before min()/max(), returns None (skip signal) on empty/invalid slices for both support and resistance branches, plus a final defensive stop_loss<=0 guard. 14 new unit tests cover zero-price, empty-slice, NaN, and negative-price cases plus a valid-data regression. CI green (lint/test/security/codecov), mergeStateStatus CLEAN, mergeable MERGEABLE, zero unresolved review threads. Matches AC directly.

Project board status: **In review** (awaiting merge phase).

**Comment by yurisa2** [comment]
### BMAD lifecycle: merge complete

The change has been merged. PR: https://github.com/PetroSa2/petrosa-bot-ta-analysis/pull/300

Project board status: **Done**.

**Comment by yurisa2** [comment]
### BMAD lifecycle: finalization complete

PR #300 merged (squash, SHA 9e3ffe3b52239131c3d8a9f7af5490146656f54c). Fix adds OrderFlowImbalanceStrategy._safe_level(), which filters non-positive/NaN prices out of the support/resistance lookback window before min()/max() and returns None (skip signal) on an empty/invalid window, plus a final stop_loss<=0 guard before Signal construction. 14 new tests cover zero-price, empty-slice, NaN, and negative-price edge cases plus a valid-data regression, mirroring existing tests/strategies/ patterns. Board moved to Done. No k8s deploy verification applies (impl_repo=petrosa-bot-ta-analysis, not petrosa_k8s).
